python/ray/serve/autoscaling_policy.py

```diff
@@ -92,47 +89,15 @@ def _calculate_desired_num_replicas(
 
 
 @PublicAPI(stability="alpha")
-class AutoscalingPolicy:
-    """Defines the interface for an autoscaling policy.
-
-    To add a new autoscaling policy, a class should be defined that provides
-    this interface. The class may be stateful, in which case it may also want
-    to provide a non-default constructor. However, this state will be lost when
-    the controller recovers from a failure.
-    """
-
-    __metaclass__ = ABCMeta
-
-    def __init__(self, config: AutoscalingConfig):
-        """Initialize the policy using the specified config dictionary."""
-        self.config = config
-
-    @abstractmethod
-    def get_decision_num_replicas(
-        self,
-        curr_target_num_replicas: int,
-        current_num_ongoing_requests: List[float],
-        current_handle_queued_queries: float,
-    ) -> int:
-        """Make a decision to scale replicas.
-
-        Arguments:
-            current_num_ongoing_requests: List[float]: List of number of
-                ongoing requests for each replica.
-            curr_target_num_replicas: The number of replicas that the
-                deployment is currently trying to scale to.
-            current_handle_queued_queries : The number of handle queued queries,
-                if there are multiple handles, the max number of queries at
-                a single handle should be passed in
-
-        Returns:
-            int: The new number of replicas to scale to.
-        """
-        return curr_target_num_replicas
-
-
-@PublicAPI(stability="alpha")
-class ReplicaQueueLengthAutoscalingPolicy(AutoscalingPolicy):
+def replica_queue_length_autoscaling_policy(
+    curr_target_num_replicas: int,
+    current_num_ongoing_requests: List[float],
+    current_handle_queued_queries: float,
+    config: Optional[AutoscalingConfig],
+    capacity_adjusted_min_replicas: int,
+    capacity_adjusted_max_replicas: int,
+    policy_state: Dict[str, Any],
```

Comment:
This seems quite verbose and not very flexible for us to add new sources.

Let's create a data class incorporating all of that (something along the lines of `AutoscalingPolicyInput`)

Reply:
Yes, the upcoming PR will add an `AutoscalingContext` class to wrap all the inputs :)